### PR TITLE
Tutorial: Spying on properties

### PR DESCRIPTION
**Pull request by elliot-nelson**
### SUMMARY

Document how to spy on properties, how to create a spy object with properties, and how to correctly change spies on properties after creating them.

### DETAILS

Fairly simple, but I've seen the question multiple times -- hopefully this will make it more discoverable.

Related to https://github.com/jasmine/jasmine/pull/1719, https://github.com/jasmine/jasmine/issues/1442.

### CODE REVIEW

File: _tutorials/spying_on_properties.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+You can create a spy object with several properties on it quickly by passing an array or hash of properties as
+a third argument to `createSpyObj`. In this case you won't have a reference to the created spies, so if you
+need to change their spy strategies later, you can use the `Object.getOwnPropertyDescriptor` approach.
+
+```javascript
+it("creates a spy object with properties", function() {
+  let obj = createSpyObj("myObject", {}, { x: 3, y: 4 });
+  expect(obj.x).toEqual(3);
+
+  Object.getOwnPropertyDescriptor(obj, "x", "get").and.returnValue(7);
```
Comment (Ksisu) [inline _tutorials/spying_on_properties.md:56]: `Object.getOwnPropertyDescriptor` has only two params.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getOwnPropertyDescriptor

When i tried this i got en error:

> Cannot read property 'returnValue' of undefined

I think it should be:
```
    Object.getOwnPropertyDescriptor(obj, "x").get.and.returnValue(7);
```

Comment (elliot-nelson) [reply _tutorials/spying_on_properties.md:56]: 🤦 Thanks for catching! Will put up a correction.

Comment (souptikmakarov) [reply _tutorials/spying_on_properties.md]: Object.getOwnPropertyDescriptor(obj, "x").get.and.returnValue(7); gives error as `Property 'and' does not exist on type '() => any'.`

Comment (Ksisu) [reply _tutorials/spying_on_properties.md]: @souptikmakarov it's a typescript issue, quick "fix" to add `// @ts-ignore` before the row. Like this:
```
      const obj = createSpyObj('myObject', {}, { x: 3, y: 4 });
      expect(obj.x).toEqual(3);
      // @ts-ignore
      Object.getOwnPropertyDescriptor(obj, 'x').get.and.returnValue(7);
      expect(obj.x).toEqual(7);
```
Or just cast to any (i think it's uglier):
```
      (Object.getOwnPropertyDescriptor(obj, 'x').get as any).and.returnValue(7);
```

**Pull request merged**